Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift

```diff
@@ -93,11 +94,11 @@ final class NetworkProtectionConnectionTester {
     // MARK: - Test result handling
 
     private var failureCount = 0
-    private let resultHandler: @MainActor (Result) -> Void
+    private let resultHandler: @MainActor (Result, Bool) -> Void
```

Comment:
The bool is for letting the callback now if this is the startup test done for an on-demand startup.

Reply:
That's probably okay, I was thinking we could perhaps make a typealias out of it or evenprovide a struct to encapsulate everything cleanly (since looking at this now, it's not clear what the Bool is for), but I'm not going to block on it.

Perhaps even just a doc comment is the way to go.